🌐 Add Chinese translation for Tutorial - Middleware

lpdswing opened the pull request.

lpdswing committed:
add chinese translate for tutorial-middleware

lpdswing commented:
@tiangolo  I don't know why occured this error when build doc .
![image](https://user-images.githubusercontent.com/20874036/98640888-cf35a300-2365-11eb-97aa-123739533422.png)

tiangolo committed:
🔀 Merge master to include fix for CI

tiangolo committed:
🔀 Merge master to fix CI build, strike 1 ⚾

tiangolo commented:
Hey @lpdswing ! Thanks for letting me know.

The error had nothing to do with you, it was caused by a recent change I made when I enabled the new features of [Material for MkDocs Insiders](https://squidfunk.github.io/mkdocs-material/insiders/), but I incorrectly handled the install in PRs from forks.

I just fixed it and updated your branch, so it's now passing and ready for review :tada:

lpdswing commented:
OK,thx发自我的华为手机-------- 原始邮件 --------发件人： Sebastián Ramírez <notifications@github.com>日期： 2020年11月11日周三 凌晨3:56收件人： tiangolo/fastapi <fastapi@noreply.github.com>抄送： lpdswing <lpdswing@qq.com>, Mention <mention@noreply.github.com>主    题： Re: [tiangolo/fastapi] Add chinese translate for tutorial-middleware (#2334)

Hey @lpdswing ! Thanks for letting me know.

The error had nothing to do with you, it was caused by a recent change I made when I enabled the new features of Material for MkDocs Insiders, but I incorrectly handled the install in PRs from forks.

I just fixed it and updated your branch, so it's now passing and ready for review 🎉



—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.

waynerv commented on docs/zh/docs/tutorial/middleware.md:
```diff
@@ -0,0 +1,61 @@
+
+你可以向 **FastAPI** 应用添加中间件.
+
+"middleware"是一个函数,它在处理每个特定*路径操作*的**request**前工作,并且在每个**response**返回之前工作.
```
「中间件」是一个函数，它在每个**请求**被特定的*路径操作*处理之前以及每个**响应**被返回之前工作.

waynerv commented on docs/zh/docs/tutorial/middleware.md:
```diff
@@ -0,0 +1,61 @@
+
+"middleware"是一个函数,它在处理每个特定*路径操作*的**request**前工作,并且在每个**response**返回之前工作.
+
+* 它接收你的应用程序的每一个**request**.
```
`request` and `response` should be translated.

waynerv commented on docs/zh/docs/tutorial/middleware.md:
```diff
@@ -0,0 +1,61 @@
+
+* 它接收你的应用程序的每一个**request**.
+* 然后它可以对这个**request**做一些事情或者执行任何需要的代码.
+* 然后它将**request**传递给应用程序的其他部分 (通过某种*路径操作*).
```
然后它将**请求**传递给应用程序的其他部分（某个*路径操作*）进行处理。

waynerv commented on docs/zh/docs/tutorial/middleware.md:
```diff
@@ -0,0 +1,61 @@
+* 它接收你的应用程序的每一个**request**.
+* 然后它可以对这个**request**做一些事情或者执行任何需要的代码.
+* 然后它将**request**传递给应用程序的其他部分 (通过某种*路径操作*).
+* 然后它获取应用程序生产的**response** (通过某种*路径操作*).
```
然后它获取应用程序（某个*路径操作*）生成的**响应** 。

waynerv commented on docs/zh/docs/tutorial/middleware.md:
```diff
@@ -0,0 +1,61 @@
+* 然后它返回这个 **response**.
+
+!!! note "技术细节"
+    如果你依赖了 `yield`, 退出代码将在执行中间件*后*执行.
```
如果你有使用了 `yield` 关键字的依赖，依赖中的退出代码将在执行中间件*之后*执行。

waynerv commented on docs/zh/docs/tutorial/middleware.md:
```diff
@@ -0,0 +1,61 @@
+!!! note "技术细节"
+    你也可以使用 `from starlette.requests import Request`.
+
+    **FastAPI** 为开发者提供了方便. 但是你也可以直接使用Starlette.
```
**FastAPI** 出于方便开发者的原因提供了该对象，但其实它直接来自于 Starlette。

waynerv commented on docs/zh/docs/tutorial/middleware.md:
```diff
@@ -0,0 +1,61 @@
+
+## 其他中间件
+
+你可以稍后再 [Advanced User Guide: Advanced Middleware](../advanced/middleware.md){.internal-link target=_blank}阅读更过关于中间件的教程.
```
你可以稍后在 [Advanced User Guide: Advanced Middleware](../advanced/middleware.md){.internal-link target=_blank}阅读更多关于中间件的教程。

lpdswing committed:
fix some mistake

RunningIkkyu commented on docs/zh/docs/tutorial/middleware.md:
```diff
@@ -0,0 +1,61 @@
+
+要创建中间件你可以在函数的顶部使用装饰器 `@app.middleware("http")`.
+
+中间件参数接收:
```
中间件函数接收如下参数：

RunningIkkyu commented on docs/zh/docs/tutorial/middleware.md:
```diff
@@ -0,0 +1,61 @@
+* 然后它返回这个 **响应**.
+
+!!! note "技术细节"
+    如果你使用了 `yield`关键字依赖, 依赖中的退出代码将在执行中间件*后*执行.
```
Leave a space after `yield`.

lpdswing replied:
删除空格?

RunningIkkyu replied:
`yield` 后面加个空格

RunningIkkyu commented on docs/zh/docs/tutorial/middleware.md:
```diff
@@ -0,0 +1,61 @@
+
+* `request`.
+* 一个函数 `call_next` 它将接收 `request` 作为参数.
+    * 这个函数将 `request` 传递个相应的 *路径操作*.
```
传递给

lpdswing replied:
thx

RunningIkkyu commented on docs/zh/docs/tutorial/middleware.md:
```diff
@@ -0,0 +1,61 @@
+!!! note "技术细节"
+    你也可以使用 `from starlette.requests import Request`.
+
+    **FastAPI** 为了开发者方便提供了该对象. 但其实它直接来自于Starlette.
```
Leave a space before `Starlette`.

RunningIkkyu commented on docs/zh/docs/tutorial/middleware.md:
```diff
@@ -0,0 +1,61 @@
+!!! tip
+    请记住可以 <a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers" class="external-link" target="_blank">用'X-' 前缀</a>添加专有自定义请求头.
+
+    但是如果你想让浏览器中的客户端看到你的自定义请求头, 你需要把它们加到 CORS 配置 ([CORS (Cross-Origin Resource Sharing)](cors.md){.internal-link target=_blank}) 的 `expose_headers` 参数中,在 <a href="https://www.starlette.io/middleware/#corsmiddleware" class="external-link" target="_blank">Starlette's CORS docs</a>文档中.
```
Leave spaces for `Starlette's CORS docs`.

RunningIkkyu commented on docs/zh/docs/tutorial/middleware.md:
```diff
@@ -0,0 +1,61 @@
+
+### 在 `response` 的前和后
+
+在任何*路径操作*收到`request`前,可以添加要和请求一起运行的代码.
```
Wrap enlish words with a space.

lpdswing committed:
fix err

lpdswing committed:
Merge branch 'master' of https://github.com/tiangolo/fastapi into lang_zh/tutorial-middleware

waynerv commented on docs/zh/docs/tutorial/middleware.md:
```diff
@@ -0,0 +1,61 @@
+
+## 其他中间件
+
+你可以稍后再 [Advanced User Guide: Advanced Middleware](../advanced/middleware.md){.internal-link target=_blank}阅读更多关于中间件的教程.
```
再 -> 在

waynerv commented on docs/zh/docs/tutorial/middleware.md:
```diff
@@ -0,0 +1,61 @@
+
+也可以在*响应*生成但是返回之前添加代码.
+
+例如你可以添加自定义请求头 `X-Process-Time` 包含以秒为单位的接收请求和生成相应的时间:
```
相应 -> 响应

waynerv reviewed (changes requested):
You should also translate all the punctuation marks.

tiangolo commented:
Awesome! Thank you @lpdswing for your contribution! :cake: :rocket: 

And thanks @waynerv, @RunningIkkyu, and @maoyibo for the reviews :bow: :coffee: 

---

I'm going to wait for the punctuation fixes and then the approval by @waynerv before merging :nerd_face: :memo:

lpdswing committed:
Merge branch 'lang_zh/tutorial-middleware' of https://github.com/lpdswing/fastapi; branch 'master' of https://github.com/tiangolo/fastapi into lang_zh/tutorial-middleware

lpdswing committed:
fix error

waynerv reviewed (approved):
@tiangolo I think this PR is ready for merging.

tiangolo commented:
Excellent, thanks for the confirmation @waynerv ! :rocket:
